L1Trigger/L1TTrackerDTC/src/Module.cc

```diff
@@ -0,0 +1,113 @@
+    sin_ = sin(tilt_);
+    cos_ = cos(tilt_);
+    layerId_ = barrel_ ? trackerTopology->layer(detId) : trackerTopology->tidWheel(detId) + 10;  // layer id [1-6,11-15]
+
+    signRow_ = signbit(deltaPhi(plane.rotation().x().phi() - pos0.phi()));  // TTStub row needs flip of sign
+    signCol_ = !barrel_ && !side_;                                          // TTStub col needs flip of sign
+    signBend_ = barrel_ || (!barrel_ && side_);                             // TTStub bend needs flip of sign
+
+    if (settings->dataFormat() != "Hybrid")
+      return;
+
+    SettingsHybrid* format = settings->hybrid();
+
+    const vector<int>& numRingsPS = format->numRingsPS();
+    const vector<double>& diskZs = format->diskZs();
+    const vector<double>& layerRs = format->layerRs();
+    const vector<double>& numTiltedLayerRings = format->numTiltedLayerRings();
+    const vector<double>& windowSizeBarrelLayers = format->windowSizeBarrelLayers();
+    const vector<vector<double> >& windowSizeTiltedLayerRings = format->windowSizeTiltedLayerRings();
+    const vector<vector<double> >& windowSizeEndcapDisksRings = format->windowSizeEndcapDisksRings();
+    const vector<vector<double> >& bendEncodingsPS = format->bendEncodingsPS();
+    const vector<vector<double> >& bendEncodings2S = format->bendEncodings2S();
+    const vector<vector<int> >& layerIdEncodings = format->layerIdEncodings();
+
+    const bool psModule = trackerGeometry->getDetectorType(detId) == TrackerGeometry::ModuleType::Ph2PSP;
+
+    if (barrel_ && psModule)
+      type_ = SettingsHybrid::barrelPS;
+    if (barrel_ && !psModule)
+      type_ = SettingsHybrid::barrel2S;
+    if (!barrel_ && psModule)
+      type_ = SettingsHybrid::diskPS;
+    if (!barrel_ && !psModule)
+      type_ = SettingsHybrid::disk2S;
+
+    // getting windows size for bend encoding
+    const int type = trackerTopology->tobSide(detId);  // 1 for tilted-, 2 for tilted+, 3 for flat
+
+    int ladder = barrel_ ? trackerTopology->tobRod(detId) : trackerTopology->tidRing(detId);
+    if (barrel_ && type == 1)
+      ladder = 1 + numTiltedLayerRings.at(layerId_) -
+               ladder;  // Corrected ring number, bet 0 and barrelNTilt.at(layer), in ascending |z|
+
+    double windowSize =
+        barrel_ ? windowSizeBarrelLayers.at(layerId_) : windowSizeEndcapDisksRings.at(layerId_ - 10).at(ladder);
+    if (barrel_ && type != 3)
+      windowSize = windowSizeTiltedLayerRings.at(layerId_).at(ladder);
+    const int ws = windowSize / format->baseWindowSize();
+
+    // getting bend encoding
+    bendEncoding_ =
+        psModule ? bendEncodingsPS.at(ws) : bendEncodings2S.at(ws);  // index = encoded bend, value = decoded bend
+
+    // encoding for 2S endcap radii
+    decodedR_ = type_ == SettingsHybrid::disk2S ? numColumns_ * (ladder - numRingsPS.at(layerId_ - 11)) : 0;
+
+    // r and z offsets
+    offsetR_ = barrel_ ? layerRs.at(layerId_ - 1) : 0.;
+    offsetZ_ = barrel_ ? 0. : diskZs.at(layerId_ - 11);
```

Comment:
magic number 11 should be named constant

Reply:
done